File: pRobot169/python/webpost07-4.py
import re
import os
import sys
import requests
from requests.packages.urllib3.exceptions import InsecureRequestWarning

# Suppress only the single InsecureRequestWarning from urllib3 needed for requests
requests.packages.urllib3.disable_warnings(InsecureRequestWarning)

gBrdName = sys.argv[1]
gDate = sys.argv[2]
file_uid_pass = sys.argv[3]

if not (re.match(r'^\w+$', gBrdName) and re.match(r'^\d\d/\d\d$', gDate) and os.path.isfile(file_uid_pass) and os.path.isfile("stat.txt")):
    print("""Usage: python {} boardname date uid_pass_file
    The form of parameter 'date' is 'mm/dd'.
    Please execute this program in a directory with 'stat.txt'.""".format(sys.argv[0]))
    sys.exit()

session = requests.Session()
session.cookies.load(ignore_discard=True, filename="./baha_cookie.txt")

# --- logout ----------------------------------------
req = requests.Request(
    method='HEAD',
    url='https://user.gamer.com.tw/logout.php'
)
prepped = session.prepare_request(req)
resp = session.send(prepped, verify=False)
print(resp.text)
# ===================================================
sys.exit()

File: pRobot169/perl/webpost07-4.pl
#!/usr/bin/perl
#

use strict;
my $gBrdName = shift;
my $gDate = shift;
my $file_uid_pass = shift;

if (!($gBrdName =~ /^\w+$/ && $gDate =~ m#^\d\d/\d\d$#
        && -f $file_uid_pass && -f "stat.txt" )) {
    print <<USAGE;
Usage: perl $0 boardname date uid_pass_file
    The form of parameter 'date' is 'mm/dd'.
    Please execute this program in a directory with 'stat.txt'.
USAGE
    exit;
}

use LWP::UserAgent;
use HTTP::Cookies;
use HTTP::Request::Common qw(GET POST);

my $ua = LWP::UserAgent->new;
my $cookie_jar = HTTP::Cookies->new(
    ignore_discard => 1, file => "./baha_cookie.txt");
$cookie_jar->load;
$ua->cookie_jar($cookie_jar);
$ua->protocols_allowed(undef);

foreach (1) {
    # --- logout ----------------------------------------
    my $req = HTTP::Request->new(
        HEADER => 'https://user.gamer.com.tw/logout.php');
    print $ua->request($req)->as_string;
    # ===================================================
    last;
}
exit;